api/repositories/pod_repository.go

```diff
@@ -47,6 +47,54 @@ func (r *PodRepo) listPods(ctx context.Context, authInfo authorization.Info, lis
 	return podList.Items, nil
 }
 
+func (r *PodRepo) DeletePod(ctx context.Context, authInfo authorization.Info, appRevision string, appGUID string, appSpaceGUID string, instanceID string, processType string) error {
+	labelSelector, err := labels.ValidatedSelectorFromSet(map[string]string{
+		"korifi.cloudfoundry.org/app-guid":     appGUID,
+		"korifi.cloudfoundry.org/version":      appRevision,
+		"korifi.cloudfoundry.org/process-type": processType,
+	})
+	if err != nil {
+		return fmt.Errorf("failed to build labelSelector: %w", apierrors.FromK8sError(err, PodResourceType))
+	}
+	listOpts := client.ListOptions{Namespace: appSpaceGUID, LabelSelector: labelSelector}
+
+	// find the pod name from the app
+	pods, err := r.listPods(ctx, authInfo, listOpts)
+	if err != nil {
+		return fmt.Errorf("failed to list pods: %w", apierrors.FromK8sError(err, PodResourceType))
+	}
+	if len(pods) == 0 {
+		return fmt.Errorf("no pods found for app and process: %w", apierrors.FromK8sError(err, PodResourceType))
+	}
+	pod := corev1.Pod{}
```

Comment:
This could be `var *corev1.Pod` and in this case we would not need a `found` boolean, since we could just nil-check the pointer.

Reply:
cool idea - done